### PR TITLE
Whitelisting for the Noodle security sandbox

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/main/resources/component-biz.conf [hunk 1]
```diff
@@ -130,6 +130,25 @@ sirius.frameworks {
...
+            "sirius.db.mixing.BaseEntity.getUniqueName" = true
+            "sirius.db.mixing.BaseEntity.isNew" = true
+            "sirius.db.mixing.types.BaseEntityRef.fetchValue" = true
+            "sirius.db.mixing.types.BaseEntityRef.getId" = true
+            "sirius.db.mixing.types.BaseEntityRef.getIdAsString" = true
+            "sirius.db.mixing.types.BaseEntityRef.is" = true
+            "sirius.db.mixing.types.BaseEntityRef.isFilled" = true
+            "sirius.db.mixing.types.SafeList.contains" = true
+            "sirius.db.mixing.types.SafeList.isEmpty" = true
+            "sirius.db.mixing.types.SafeList.isFilled" = true
```
Comment (sabieber) [inline src/main/resources/component-biz.conf:147]: Why arent these done via annotations in db?

Comment (jmuscireum) [reply src/main/resources/component-biz.conf:147]: Because sirius-db doesn't have sirius-web as dependency

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/model/LoginData.java [hunk 2]
```diff
@@ -424,7 +426,7 @@ public String getPasswordHash() {
         return passwordHash;
     }
 
-
+    @NoodleSandbox(NoodleSandbox.Accessibility.GRANTED)
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/model/LoginData.java:429]: Is this too spicy?

File: src/main/resources/component-biz.conf (hunk 1 above)
Comment (idlira) [reply src/main/resources/component-biz.conf:147]: Yeah, that's a bit ugly... 
If Sacha's approach with provided dependencies do not work, reactivate the _OneSirius_ conversation? 😇

File: src/main/java/sirius/biz/model/LoginData.java (hunk 2 above)
Comment (qw3ry) [reply src/main/java/sirius/biz/model/LoginData.java:429]: Maybe add an "getGeneratedPasswordIfDisplayed()" that returns the password only if "isDisplayGeneratedPassword()"? That should be sufficient for the use case, and adds a tiny bit security

File: src/main/java/sirius/biz/tenants/mongo/MongoTenant.java [hunk 3]
```diff
@@ -103,6 +104,7 @@ public MongoRef<MongoTenant> getParent() {
     }
 
     @Override
+    @NoodleSandbox(NoodleSandbox.Accessibility.GRANTED)
```
Comment (andyHa) [inline src/main/java/sirius/biz/tenants/mongo/MongoTenant.java:107]: put on super-method?

File: src/main/java/sirius/biz/tenants/mongo/MongoTenantAware.java [hunk 4]
```diff
@@ -48,6 +49,7 @@ public abstract class MongoTenantAware extends MongoBizEntity implements TenantA
     private final MongoRef<MongoTenant> tenant = MongoRef.writeOnceOn(MongoTenant.class, MongoRef.OnDelete.REJECT);
 
     @Override
+    @NoodleSandbox(NoodleSandbox.Accessibility.GRANTED)
```
Comment (andyHa) [inline src/main/java/sirius/biz/tenants/mongo/MongoTenantAware.java:52]: s.a.

File: src/main/java/sirius/biz/tenants/mongo/MongoTenant.java (hunk 3 above)
Comment (jmuscireum) [reply src/main/java/sirius/biz/tenants/mongo/MongoTenant.java:107]: The annotations currently don't work on interfaces

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by mko-sci — APPROVED** [review]

**Pull request merged**